dockstore-webservice/src/main/java/io/openapi/api/impl/ToolsApiServiceImpl.java

```diff
@@ -266,47 +273,47 @@ public boolean canExamine(User user, Entry entry) {
     }
 
     @Override
-    public Response toolsIdVersionsVersionIdTypeDescriptorGet(String type, String id, String versionId, SecurityContext securityContext,
-        ContainerRequestContext value, Optional<User> user) {
-        final Optional<DescriptorLanguage.FileType> fileType = DescriptorLanguage.getOptionalFileType(type);
+    public Response toolsIdVersionsVersionIdTypeDescriptorGet(String id, DescriptorTypeWithPlain type, String versionId, SecurityContext securityContext, ContainerRequestContext value,
```

Comment:
I don't think it matters, but just making sure -- the fact that the parameter order changed, in addition to the type of the type parameter, won't break anything, right? I think these are delegated endpoints, so it shouldn't matter, but I want to be sure the old CLI wil still work, for example.

Reply:
Short version: I think so, I'll add an ticket to test this. I think we have regression tests that should kick in on the release branch but might be nice to have someone explicitly look for them. 

Long version: Yeah, it's a dumb part of how the mapping works between REST, Java
Like if you have `GET /ga4gh/trs/v2/tools/{id}/versions/{version_id}/{type}/descriptor` or `https://dockstore.org/api/ga4gh/trs/v2/tools/12345/versions/v1/CWL/descriptor`
You know that id, version_id, and type are the names of the parameters 12345, v1, and CWL respectively so order is irrelevant. But the switch to OpenAPI must have generated some functions in a different order which confuses Java because Java lacks named parameters https://en.wikipedia.org/wiki/Named_parameter 
In a similar vein, both string and enum map to the same thing. For example a free-form `CWL` is the same as a enum `CWL` if I understand properly.

Reply:
Continues with https://ucsc-cgl.atlassian.net/browse/SEAB-5039